packages/wonder-blocks-core/components/clickable.js

```diff
@@ -0,0 +1,141 @@
+import type {StyleType} from "../util/types.js";
+
+type Props = {|
+    "aria-label": string,
+    children: (state: State) => React.Node,
+    disabled: boolean,
+    href?: string,
+    onClick?: (e: SyntheticEvent<>) => mixed,
+    role?: ClickableRole,
+    skipClientNav?: boolean,
+    style?: StyleType,
+    testId?: string,
+|};
+
+const StyledAnchor = addStyle<"a">("a");
+const StyledButton = addStyle<"button">("button");
+const StyledLink = addStyle<typeof Link>(Link);
+
+export default class Clickable extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        "aria-label": "",
+    };
+
+    render() {
+        const {router} = this.context;
+        const {
+            "aria-label": ariaLabel, // eslint-disable-line react/prop-types
+            children,
+            disabled,
+            href,
+            onClick,
+            role,
+            skipClientNav,
+            style,
+            testId,
+        } = this.props;
+
+        const ClickableBehavior = getClickableBehavior(
+            href,
+            skipClientNav,
+            this.context.router,
+        );
+
+        return (
+            <ClickableBehavior
+                disabled={disabled}
+                href={href}
+                onClick={onClick}
+                role={role}
+            >
+                {(state, handlers) => {
+                    const commonProps = {
+                        "aria-label": ariaLabel,
+                        "data-test-id": testId,
+                        role: role,
+                        style: [styles.reset, style],
+                        ...handlers,
+                    };
+
+                    const content = children(state);
+
+                    if (href && !disabled) {
+                        return router && !skipClientNav ? (
+                            <StyledLink
+                                {...commonProps}
+                                to={href}
+                                aria-disabled={disabled ? "true" : undefined}
+                            >
+                                {content}
+                            </StyledLink>
+                        ) : (
+                            <StyledAnchor
+                                {...commonProps}
+                                href={href}
+                                aria-disabled={disabled ? "true" : undefined}
+                            >
+                                {content}
+                            </StyledAnchor>
+                        );
+                    } else {
+                        return (
+                            <StyledButton
+                                type="button"
+                                {...commonProps}
+                                disabled={disabled}
+                            >
+                                {content}
+                            </StyledButton>
+                        );
```

Comment:
Tempted to tidy this up by having some seperate methods like `maybeRenderStyledButton`, etc. and then have them control the conditions of if they render. Or at least have `renderStyledButton` and have it extract the props it needs. I think it would make the code a little easier to follow.